File: app/src/main/java/com/mehmetakiftutuncu/muezzin/adapters/viewholders/DistrictViewHolder.java
package com.mehmetakiftutuncu.muezzin.adapters.viewholders;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.mehmetakiftutuncu.muezzin.R;
import com.mehmetakiftutuncu.muezzin.fragments.DistrictSelectionFragment;
import com.mehmetakiftutuncu.muezzin.models.District;

/**
 * Created by akif on 11/05/16.
 */
public class DistrictViewHolder extends RecyclerView.ViewHolder {
    private View districtItemLayout;
    private TextView textViewName;

    private DistrictSelectionFragment.OnDistrictSelectedListener onDistrictSelectedListener;

    public DistrictViewHolder(View districtItemLayout, DistrictSelectionFragment.OnDistrictSelectedListener onDistrictSelectedListener) {
        super(districtItemLayout);

        this.districtItemLayout = districtItemLayout;
        this.onDistrictSelectedListener = onDistrictSelectedListener;

        textViewName = (TextView) districtItemLayout.findViewById(R.id.textView_item_district_name);
    }

    public void setFrom(final District district) {
        textViewName.setText(district.name);

        districtItemLayout.setOnClickListener(v -> onDistrictSelectedListener.onDistrictSelected(district));
    }
}

File: app/src/main/java/com/mehmetakiftutuncu/muezzin/adapters/viewholders/DistrictViewHolder.kt
package com.mehmetakiftutuncu.muezzin.adapters.viewholders

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.mehmetakiftutuncu.muezzin.R
import com.mehmetakiftutuncu.muezzin.fragments.SelectionFragment
import com.mehmetakiftutuncu.muezzin.models.District

class DistrictViewHolder(private val view: View,
                         private val listener: SelectionFragment.OnSelectedListener<District>): RecyclerView.ViewHolder(view) {
    private val textViewName: TextView = view.findViewById(R.id.textView_item_district_name)

    fun set(district: District) {
        textViewName.text = district.name

        view.setOnClickListener {
            listener.onSelected(district)
        }
    }
}